app/gui/language/span-tree/src/generate.rs

```diff
@@ -153,27 +153,41 @@ struct ApplicationBase<'a> {
     function_name: Option<&'a str>,
     /// True when Ast uses method notation to pass this as an invocation target.
     has_target:    bool,
+    /// The ast id of the call expression.
+    call_id:       Option<Id>,
+    /// The id of the call target operand. The target can present with not assigned ast id.
+    /// In that case the `target_id` is `None` an `has_target` is `true`.
+    target_id:     Option<Id>,
 }
 
 impl<'a> ApplicationBase<'a> {
-    fn new(ast: &'a Ast) -> Self {
-        if let Some(chain) = ast::opr::as_access_chain(ast) {
-            let get_name = || -> Option<&'a str> {
-                let crumbs = chain.enumerate_operands().last()??.crumbs;
-                let ast = ast.get_traversing(&crumbs).ok()?;
-                ast::identifier::name(ast)
-            };
-            ApplicationBase { function_name: get_name(), has_target: true }
+    fn new(ast: &'a Ast, call_id: Option<Id>, external_target_id: Option<Id>) -> Self {
+        if let Some(infix) = GeneralizedInfix::try_new(ast)
+            .filter(|infix| infix.name() == ast::opr::predefined::ACCESS)
+        {
+            let target = infix.target_operand();
+            let target_id = target.as_ref().and_then(|t| t.arg.id);
+            let function = ast.get(&infix.argument_crumb()).ok();
+            let function_name = function.and_then(ast::identifier::name);
+            ApplicationBase { function_name, call_id, target_id, has_target: true }
```

Comment:
this is pretty complex one now. It should have docs + maybe refactoring to helper, well-named functions.